pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator_test.go

```diff
@@ -350,6 +367,829 @@ var _ = Describe("VirtualMachine Mutator", func() {
+				},
+			),
+		)
+		DescribeTable("should infer defaults from DataVolume with annotations", func(instancetypeMatcher, expectedInstancetypeMatcher *v1.InstancetypeMatcher, preferenceMatcher, expectedPreferenceMatcher *v1.PreferenceMatcher) {
+			vm.Spec.Instancetype = instancetypeMatcher
+			vm.Spec.Preference = preferenceMatcher
+
+			dvWithAnnotations = &v1beta1.DataVolume{
+				ObjectMeta: k8smetav1.ObjectMeta{
+					Name:      "dvWithAnnotations",
+					Namespace: vm.Namespace,
+					Annotations: map[string]string{
+						instancetype.DefaultInstancetypeAnnotation:     defaultInferedName,
+						instancetype.DefaultInstancetypeKindAnnotation: defaultInferedKind,
+						instancetype.DefaultPreferenceAnnotation:       defaultInferedName,
+						instancetype.DefaultPreferenceKindAnnotation:   defaultInferedKind,
+					},
+				},
+				TypeMeta: k8smetav1.TypeMeta{
+					Kind: "DataVolume",
+				},
+				Spec: v1beta1.DataVolumeSpec{
+					Source: &v1beta1.DataVolumeSource{
+						PVC: &v1beta1.DataVolumeSourcePVC{
+							Name:      pvc.Name,
+							Namespace: pvc.Namespace,
+						},
+					},
+				},
+			}
+			_, err := virtClient.CdiClient().CdiV1beta1().DataVolumes(vm.Namespace).Create(context.Background(), dvWithAnnotations, k8smetav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
+				Name: inferVolumeName,
+				VolumeSource: v1.VolumeSource{
+					DataVolume: &v1.DataVolumeSource{
+						Name: dvWithAnnotations.Name,
+					},
+				},
+			})
+
+			vmSpec, _ := getVMSpecMetaFromResponse()
+			Expect(vmSpec.Instancetype).To(Equal(expectedInstancetypeMatcher))
+			Expect(vmSpec.Preference).To(Equal(expectedPreferenceMatcher))
+		},
+			Entry("for InstancetypeMatcher",
+				&v1.InstancetypeMatcher{
+					InferFromVolume: inferVolumeName,
+				},
+				&v1.InstancetypeMatcher{
+					Name: defaultInferedName,
+					Kind: defaultInferedKind,
+				}, nil, nil,
+			),
+			Entry("for PreferenceMatcher",
+				nil, nil,
+				&v1.PreferenceMatcher{
+					InferFromVolume: inferVolumeName,
+				},
+				&v1.PreferenceMatcher{
+					Name: defaultInferedName,
+					Kind: defaultInferedKind,
+				},
+			),
+		)
+
+		DescribeTable("should infer defaults from DataVolume, DataVolumeSourceRef", func(sourceRefName, sourceRefKind, sourceRefNamespace string, instancetypeMatcher, expectedInstancetypeMatcher *v1.InstancetypeMatcher, preferenceMatcher, expectedPreferenceMatcher *v1.PreferenceMatcher) {
+			vm.Spec.Instancetype = instancetypeMatcher
+			vm.Spec.Preference = preferenceMatcher
+			dvWithSourceRef := &v1beta1.DataVolume{
+				ObjectMeta: k8smetav1.ObjectMeta{
+					Name:      "dvWithSourceRef",
+					Namespace: vm.Namespace,
+				},
+				TypeMeta: k8smetav1.TypeMeta{
+					Kind: "DataVolume",
+				},
+				Spec: v1beta1.DataVolumeSpec{
+					SourceRef: &v1beta1.DataVolumeSourceRef{
+						Name:      sourceRefName,
+						Kind:      sourceRefKind,
```

Comment:
Again, not sure if `PersistentVolumeClaim` is valid here.

Reply:
Removed.